### PR TITLE
Minor - add link to collect

### PR DESCRIPTION
**Pull request by cijothomas**
Fixes #

## Changes

Please provide a brief description of the changes here. Update the
`CHANGELOG.md` for non-trivial changes. If `CHANGELOG.md` is updated,
also be sure to update `spec-compliance-matrix.md` if necessary.

Related issues #

Related [oteps](https://github.com/open-telemetry/oteps) #

### CODE REVIEW

File: specification/metrics/api.md [hunk 1]
```diff
@@ -254,7 +254,7 @@ asynchronous:
 
 * Asynchronous instruments (e.g. [Asynchronous Gauge](#asynchronous-gauge)) give
   the user a way to register callback function, and the callback function will
-  only be invoked upon collection. For example, a piece of embedded software
+  only be invoked upon [collection](../sdk.md#collect). For example, a piece of embedded software
```
Comment (SergeyKanzhelev) [inline specification/metrics/api.md:257]: did you mean: 

```suggestion
  only be invoked upon [collection](sdk.md#collect). For example, a piece of embedded software
```

**Review by SergeyKanzhelev — APPROVED** [review]

File: specification/metrics/api.md [hunk 2]
```diff
@@ -254,7 +254,7 @@ asynchronous:
 
 * Asynchronous instruments (e.g. [Asynchronous Gauge](#asynchronous-gauge)) give
   the user a way to register callback function, and the callback function will
-  only be invoked upon collection. For example, a piece of embedded software
+  only be invoked upon [collection](sdk.md#collect). For example, a piece of embedded software
```
Comment (reyang) [inline specification/metrics/api.md:257]: I think the "collection" here is a general concept rather than a concrete method/function.

Comment (cijothomas) [reply specification/metrics/api.md:257]: Thats right.. There is no section which describes the general concept of "collection", so I picked the closest. 
Maybe that's a good addition to the spec, so its easy to explain "when does my async instrument callback gets executed?". Will follow up later, and abandon this PR.

**Review by bogdandrutu — APPROVED** [review]

File: specification/metrics/api.md (hunk 2 above)
Comment (cijothomas) [reply specification/metrics/api.md:257]: I think its still okay to have this. There is no other place which talks about "when does the callbacks from observable instruments gets called".

Comment (carlosalberto) [reply specification/metrics/api.md:257]: What about something along the lines of:

```
only be invoked upon collection (see the [collect](sdk.md#collect) operation in the SDK for reference)
```

**Review by carlosalberto — APPROVED** [review]

**Pull request merged**